src/window_interpreter.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Headers
+#include <vector>
+#include "window_interpreter.h"
+#include "bitmap.h"
+#include "game_map.h"
+#include "game_message.h"
+#include "game_system.h"
+#include "game_variables.h"
+#include "input.h"
+#include "player.h"
+#include "lcf/reader_util.h"
+
+namespace {
+	std::vector<std::string> CreateEmptyLines(int c) {		
+		std::vector<std::string> vars;
+		for (int i = 0; i < c; i++)
+			vars.push_back("");
+		return vars;
+	}
+}
+
+Window_Interpreter::Window_Interpreter(int ix, int iy, int iwidth, int iheight) :
+	Window_Selectable(ix, iy, iwidth, iheight) {
+	column_max = 1;
+}
+
+Window_Interpreter::~Window_Interpreter() {
+
+}
+
+void Window_Interpreter::SetStackState(bool is_ce, std::string interpreter_desc, lcf::rpg::SaveEventExecState state) {
+	this->interpreter_desc = interpreter_desc;
+	this->state = state;
+	this->is_ce = is_ce;
+}
+
+void Window_Interpreter::Refresh() {
+	stack_display_items.clear();
+
+	int max_cmd_count = 0;
+
+	for (int i = state.stack.size() - 1; i >= 0; i--) {
+		int evt_id = state.stack[i].event_id;
+		bool is_calling_ev_ce = false;
+
+		if (evt_id == 0 && i > 0) {
+			auto& prev_frame = state.stack[i - 1];
+			auto& com = prev_frame.commands[prev_frame.current_command - 1];
+			if (com.code == 12330) { // CallEvent
+				is_calling_ev_ce = true;
+				if (com.parameters[0] == 0) {
+					evt_id = com.parameters[1];
+				} else if (com.parameters[0] == 3 && Player::IsPatchManiac()) {
+					evt_id = Main_Data::game_variables->Get(com.parameters[1]);
+				} else if (com.parameters[0] == 4 && Player::IsPatchManiac()) {
+					evt_id = Main_Data::game_variables->GetIndirect(com.parameters[1]);
```

Comment:
Are you attempting to figure out the event ID here?

Actually Maniac Patch added some debug fields to the ExexFrame for this purpose: ``maniac_event_info``, ``maniac_event_id``, ``maniac_event_page_id``. We do not have this implemented and you do not have to implement everything but ``maniac_event_id`` could be useful to track the ID across maps because its not reset to 0.

Reply:
Yeah, that's the idea..., I've noticed that there are several unimplemented fields, but skipped over them because I don't have any implementation details and didn't want to invest the time right now to figure this out. 😬
So this is far from perfect right now.
But yeah, I should probably change the tracked id field there. Completely forgot that the vanilla event_id gets reset.
I'm on holiday right now and won't have much time to look over this. If you want to change something, feel free to do so.

Reply:
No Problem. Is nothing urgent Is enough when you annotate it with a fixme for now.